dts/bindings/mtd/microchip,xec-eeprom.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: Microchip on-chip EEPROM
+
+compatible: "microchip,xec-eeprom"
+
+include: [eeprom-base.yaml, pinctrl-device.yaml]
+
+properties:
+    reg:
+      required: true
+
+    girqs:
+      type: array
+      required: true
+      description: |
+        Array of GIRQ and bit position pairs for each interrupt
+        signal the block generates.
+
+    pcrs:
+      type: array
+      required: true
+      description: PS2 PCR register index and bit position
+
+    pinctrl-0:
+      required: true
+
+    pinctrl-names:
+      required: true
```

Comment:
If the EEPROM is "on-chip", how come it requires pinctrl settings?

Reply:
eeprom is part of the package accessed through the spi pins by the eeprom hardware block

Reply:
Earlier, I asked the [question along the same lines](https://github.com/zephyrproject-rtos/zephyr/pull/45983#discussion_r897310919).  
In order to use EEPROM, pins need to be configured in SPI, yet through external SPI reader (Dediprog), EEPROM contents can't be read? 

Also, does using these SPI pins for EEPROM affect EC boot image fetching from SPI (G3 flash sharing mode / SAF)?

If these 4 pins needed for EEPROM, EC can't use those for other GPIOs?

Reply:
the pins are inside the package not exposed externally

these pins are multiplexed with the internal spi pins; so MEC1727 has the internal spi and the MEC1723 has the eeprom on the pins. This pins wont affect the external SPI pins.

Yes the pins for EEPROM (MEC1723) / internal SPI (MEC1727) cannot be used as GPIOs